### DISCUSSION TITLE
Combination of domain solutions.

### DISCUSSION THREAD

**Discussion by TLWise** [idaholab/moose#17131] [Q&A Meshing] [answered]
Does Moose allow for the solution of equations (in time) in two separate domains? For example, T in Omega 1, and p in Omega 2, then another solution in a third domain Omega 3 using the solution of T, and p from the initial two solutions and then feedback into Omega 1 for another iteration of solutions?

**Comment by mangerij** [comment]
Ah, so you would like to try and use the restart system to combine two solutions in some meanginful way. 
Are the meshes different? or is it just two different problems solved on the same mesh?

**Comment by TLWise** [comment]
Different mesh domains so that the mesh deformation in one domain has no effect on the other. The first domain (Energy Eq) has no internal geometry, the second domain has physical internal geometry (Structural Eq), the third domain only needs a mesh grid sufficient to capture the combined effects of the other two mesh domains. It then implicitly feeds information via a boundary equation back into the first to account for energy deformation based on the position of the second.

Thank you for your interest in my problem.

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Reply by mangerij** [reply]
Well if the first domain is just a scalar ODE you can use the ScalarKernels system to get your solution and then add them using a AuxKernel

**Reply by TLWise** [reply]
I will solve the Naiver Stokes equation with density and viscosity properties that vary with pressure and temperature. Would the method you suggested be applicable for the NS equation in domain one?

**Comment by TLWise** [comment]
I will solve the Naiver Stokes equation with density and viscosity properties that vary with pressure and temperature. Would the method you suggested be applicable for the NS equation in domain one?

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Comment by GiudGiud** [comment, marked as answer]
You may want to leverage the multiapp system to separate each solve. 
Any specific reason you are trying to avoid coupling the equations in a single solve?

**Reply by TLWise** [reply]
I am combining the equation of motion with heat transfer and fluid flow. When the systems of equations are coupled I experience deformation of the mesh and associated numerical error for large displacements of objects interior to the heat transfer/fluid domain. When I solve the equation of motion in it's own domain there are no mesh deformations or numerical errors.

**Reply by GiudGiud** [reply]
So your options are:
- a multiapp for each domain, and coupling through boundary conditions / transfers
- separate domains are ok too. Use an interface kernel to pass information if they are physically separate too
